app/renderer/actions/Inspector.js

```diff
@@ -106,14 +106,36 @@ export function bindAppium () {
 }
 
 
+// A debounced function that calls findElement and gets info about the element
+const findElement = _.debounce(async function (strategy, selector, dispatch, getState, path) {
+  // Get the information about the element
+  let {elementId, variableName, variableType} = await callClientMethod({
+    strategy,
+    selector,
+  });
+
+    // If the element is stale, unselect and reload the source
+  if (!elementId) {
+    dispatch({type: UNSELECT_ELEMENT});
+    return await applyClientMethod({methodName: 'source'})(dispatch);
+  } 
+
+  // Set the elementId, variableName and variableType for the selected element 
+  // (check first that the selectedElementPath didn't change, to avoid race conditions)
+  if (getState().inspector.selectedElementPath === path) {
+    dispatch({type: SET_SELECTED_ELEMENT_ID, elementId, variableName, variableType});
+  }
+}, 1000);
+
 export function selectElement (path) {
-  return async (dispatch, getState) => {
+  return (dispatch, getState) => {
```

Comment:
I'd keep async

Reply:
We have a linting rule that if there's nothing that's awaited inside of a function, don't use async.